src/java.base/share/classes/java/io/ObjectInputFilter.java

```diff
@@ -719,6 +748,8 @@ public static void setSerialFilter(ObjectInputFilter filter) {
             synchronized (serialFilterLock) {
                 if (serialFilter != null) {
                     throw new IllegalStateException("Serial filter can only be set once");
+                } else if (invalidFilterMessage != null) {
```

Comment:
The `invalidFilterMessage` is a `static` `final` `String` which gets initialized during the static initialization of the class and as such doesn't get changed after that. Do you think this lock on `serialFilterLock` is necessary for this check or it can be moved outside this `synchronized` block?

Reply:
Checking `serialFilter` for non-null required the synchronization.
The check of `invalidFilterMessage` can be pulled out of the synchronization since in expected usage the method is only called once so the performance isn't significant.